plugins/auto-sizes/includes/improve-calculate-sizes.php

```diff
@@ -199,6 +200,10 @@ function auto_sizes_calculate_better_sizes( int $id, $size, string $align, int $
 
 		case 'wide':
 			$layout_width = auto_sizes_get_layout_width( 'wide' );
+			if ( str_ends_with( $layout_width, 'px' ) ) {
+				$layout_width = (int) $layout_width * $container_relative_width;
```

Comment:
Is this a safe type conversion? 🤔  I think I'd rather do `str_replace` first to remove the px. And then perhaps also consider float values.

Reply:
Good to know. Then maybe change to `(float)` for more accuracy?

Reply:
Update in https://github.com/WordPress/performance/pull/1818/commits/82d736421b69c58b952a2dd220b6a507bdb5b060